lib/bambora/client.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Bambora
+  class Client
+    attr_accessor :merchant_id, :sub_merchant_id, :api_key
+
+    def initialize(options = {})
+      options.each do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+      yield(self) if block_given?
+    end
+
+    # Summary: Payment profiles store confidential payment information. Transactions can be processed against profiles.
+    # Docs: https://dev.na.bambora.com/docs/guides/payment_profiles/
+    # Endpoint: https://api.na.bambora.com/v1/profiles
+    def profiles; end
+
+    # Summary: Create and modify payments.
+    # Note: The link below links to all apis includding profiles and tokenization. There aren't great docs explaining the
+    #       /payments endpoints alone.
+    # Docs: https://dev.na.bambora.com/docs/references/payment_APIs/
+    #       https://dev.na.bambora.com/docs/references/payment_SDKs/take_payments/?shell#
+    # Endpoint: https://api.na.bambora.com/v1/payments
+    def payments; end
+
+    # Summary: Bank Electronic Funds Transfer (CAD) and Automatic Clearing House (USD)
+    # Docs: https://dev.na.bambora.com/docs/guides/batch_payment/
+    #       https://dev.na.bambora.com/docs/references/batch_payment/
+    # Endpoint: https://api.na.bambora.com/v1/batchpayments
+    def batchpayments; end
+
+    # Summary: Statuses of batch bank-to-bank transactions.
+    # Docs: https://dev.na.bambora.com/docs/guides/batch_payment/report/
+    #       https://dev.na.bambora.com/docs/references/batch_payment_report/
+    # Endpoint: https://na.bambora.com/scripts/reporting/report.aspx
+    def batchpayment_reports; end
+
+    protected
+
+    def connection
+      @connection ||= Excon.new(ENV.fetch('BAMBORA_API_URL'), headers: headers)
```

Comment:
Might want to include in a future PR that this environment variable is required. 

If there are `sandbox` vs `production` URLs available, maybe make it a client configuration variable?

Reply:
Yeah, I was debating if we wanted this as an env variable. I'll make a ticket for this.